### PR TITLE
Add Get/Set/Remove -NsxDns cmdlets with testsuite

### PR DESCRIPTION
**Pull request by alagoutte**
No yet finish (need to add more tests.. and also statistics)

### CODE REVIEW

**Comment by alagoutte** [comment]
@nmbradford @pandom @dcoghlan can you look ? it is no yet finish... (need also to look #425 because need for some tests...)

**Comment by alagoutte** [comment]
@nmbradford it is possible to launch a jenkins test ? (and get feedback/review)

**Comment by alagoutte** [comment]
Hi @nmbradford @dcoghlan 

Can you look ? :)

**Comment by dcoghlan** [comment]
jenkins test this please

**Comment by dcoghlan** [comment]
@alagoutte - The tests failed due to the following error
`[-] Change cacheSize 256ms
        Cannot set "cacheSize" because only strings can be used as values to set XmlNode properties.
        at <ScriptBlock>, C:\jenkins\workspace\WindowsPowerNsx636\tests\integration\17.Dns.Tests.ps1: line 90
        90:             Get-NsxEdge $dnsedge1name | Get-NsxDns | Set-NsxDns -cacheSize 32`

File: module/PowerNSX.psm1 [hunk 1]
```diff
@@ -36249,5 +36284,280 @@ function Copy-NsxEdge{
...
+            } else {
+                Add-XmlElement -xmlroot  $_Dns -xmlElementName "enabled" -xmlElementText $Enabled.ToString().ToLower()
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('CacheSize') ) {
+            if ( invoke-xpathquery -node $_Dns -querymethod SelectSingleNode -Query "child::cacheSize" ) {
+                $_Dns.CacheSize = $CacheSize
+            } else {
+                Add-XmlElement -xmlroot  $_Dns -xmlElementName "cacheSize" -xmlElementText $CacheSize
```
Comment (dcoghlan) [inline module/PowerNSX.psm1]: Need to convert $CacheSize to a string using `$CacheSize.tostring()` otherwise this fails the tests on Powershell Windows.

**Comment by alagoutte** [comment]
Can you retry ?

**Comment by dcoghlan** [comment]
jenkins test this please

File: module/PowerNSX.psm1 [hunk 2]
```diff
@@ -36249,5 +36284,280 @@ function Copy-NsxEdge{
...
+            if ( invoke-xpathquery -node $_Dns -querymethod SelectSingleNode -Query "child::enabled" ) {
+                $_Dns.enabled = $Enabled.ToString().ToLower()
+            } else {
+                Add-XmlElement -xmlroot  $_Dns -xmlElementName "enabled" -xmlElementText $Enabled.ToString().ToLower()
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('CacheSize') ) {
+            if ( invoke-xpathquery -node $_Dns -querymethod SelectSingleNode -Query "child::cacheSize" ) {
+                $_Dns.CacheSize = $CacheSize
```
Comment (dcoghlan) [inline module/PowerNSX.psm1]: Sorry, I didn't pick this one up earlier, but this also needs to be changed to `$CacheSize.ToString()` to pass tests as your test is modifying the cache size from 16 to 32 which falls into this if statement

Comment (alagoutte) [reply module/PowerNSX.psm1]: Done, Can you retry ?

**Comment by dcoghlan** [comment]
jenkins test this please

**Pull request merged**